dags/als-kp-dataset-dag.py

```diff
@@ -221,103 +332,157 @@ def transform_data(data: Dict[str, Any], **context) -> List[Dict[str, Any]]:
 
     @task
     def find_duplicated_datasets(
-        transformed_items: List[Dict[str, Any]], **context
-    ) -> List[Dict[str, Any]]:
-        """Flag duplicated new C-PATH datasets.
+    transformed_items: List[Dict[str, Any]], **context
+    ) -> Dict[str, List[Dict[str, Any]]]:
+        """Apply latest and greatest version selection for ALS datasets.
```

Comment:
Since you have completely changed the goal of this function, this function should be renamed to something else.